src/entities/friendships.rs

```diff
@@ -81,23 +83,20 @@ impl FriendshipRepositoryImplementation for FriendshipsRepository {
         &self.db_connection
     }
 
-    fn create_new_friendships_query<'a>(
-        &self,
-        addresses: (&'a str, &'a str),
-    ) -> Query<'a, Postgres, sqlx::postgres::PgArguments> {
-        let (address1, address2) = addresses;
-        sqlx::query("INSERT INTO friendships(id, address_1, address_2) VALUES($1,$2, $3);")
-            .bind(Uuid::parse_str(generate_uuid_v4().as_str()).unwrap())
-            .bind(address1)
-            .bind(address2)
-    }
-
     async fn create_new_friendships<'a>(
         &'a self,
         addresses: (&'a str, &'a str),
         transaction: Option<Transaction<'a, Postgres>>,
-    ) -> (Result<(), sqlx::Error>, Option<Transaction<'a, Postgres>>) {
-        let query = self.create_new_friendships_query(addresses);
+    ) -> (Result<Uuid, sqlx::Error>, Option<Transaction<'a, Postgres>>) {
+        let (address1, address2) = addresses;
+
+        let id = Uuid::parse_str(generate_uuid_v4().as_str()).unwrap();
+
+        let query =
+            sqlx::query("INSERT INTO friendships(id, address_1, address_2) VALUES($1,$2, $3);")
+                .bind(id)
+                .bind(address1)
```

Comment:
let's add the cleaning for each address here! (same for history)

Reply:
Solved it cleaning the synapse interactions